python/test_infra/ttir_builder.py

```diff
@@ -231,80 +235,194 @@ def empty(
-        assert self.get_shape(in0) == self.get_shape(
-            in1
-        ), "Elementwise `ttir.multiply` op expects inputs of same shape."
+        return self.eltwise_proxy(torch.multiply, ttir.MultiplyOp, [in0, in1])
 
-        with self._ctx, self._loc:
-            output = self.empty(self.get_shape(in0))
+    def logical_and(self, in0: Operand, in1: Operand) -> OpView:
+        return self.eltwise_proxy(torch.logical_and, ttir.LogicalAndOp, [in0, in1])
 
-            op = ttir.MultiplyOp(
-                [self._get_type(output)],
-                [in0, in1],
-                [output],
-                self._get_operand_constraint_attr(3),
-            )
+    def logical_or(self, in0: Operand, in1: Operand) -> OpView:
+        return self.eltwise_proxy(torch.logical_or, ttir.LogicalOrOp, [in0, in1])
 
-            golden = Golden(
-                torch.multiply(
-                    self._get_golden_tensor(in0), self._get_golden_tensor(in1)
-                )
-            )
-            self._store_golden(op, golden)
-            self._override_golden(output, golden)
+    def subtract(self, in0: Operand, in1: Operand) -> OpView:
+        return self.eltwise_proxy(torch.subtract, ttir.SubtractOp, [in0, in1])
 
-            return op
+    def eq(self, in0: Operand, in1: Operand) -> OpView:
+        return self.eltwise_proxy(torch.eq, ttir.EqualOp, [in0, in1])
 
-    def exp(self, in0: Operand) -> OpView:
-        """Convenience wrapper constructing `ttir.ExpOp`."""
-        with self._ctx, self._loc:
-            output = self.empty(self.get_shape(in0))
+    def ne(self, in0: Operand, in1: Operand) -> OpView:
+        return self.eltwise_proxy(torch.ne, ttir.NotEqualOp, [in0, in1])
 
-            op = ttir.ExpOp(
-                [self._get_type(output)],
-                [in0],
-                [output],
-                self._get_operand_constraint_attr(3),
-            )
+    def ge(self, in0: Operand, in1: Operand) -> OpView:
+        return self.eltwise_proxy(torch.ge, ttir.GreaterEqualOp, [in0, in1])
 
-            golden = Golden(torch.exp(self._get_golden_tensor(in0)))
-            self._store_golden(op, golden)
-            self._override_golden(output, golden)
+    def gt(self, in0: Operand, in1: Operand) -> OpView:
+        return self.eltwise_proxy(torch.gt, ttir.GreaterThanOp, [in0, in1])
 
-            return op
+    def le(self, in0: Operand, in1: Operand) -> OpView:
+        return self.eltwise_proxy(torch.le, ttir.LessEqualOp, [in0, in1])
+
+    def lt(self, in0: Operand, in1: Operand) -> OpView:
+        return self.eltwise_proxy(torch.lt, ttir.LessThanOp, [in0, in1])
+
+    def div(self, in0: Operand, in1: Operand) -> OpView:
+        return self.eltwise_proxy(torch.div, ttir.DivOp, [in0, in1])
+
+    def maximum(self, in0: Operand, in1: Operand) -> OpView:
+        return self.eltwise_proxy(torch.maximum, ttir.MaximumOp, [in0, in1])
+
+
+def compile_as_mlir_module(
```

Comment:
Any reason to have this decorator defined in 2 files? Should we remove one of them?

Reply:
will remove, thanks for the catch